ledger/txtail.go

```diff
@@ -387,8 +402,5 @@ func (t *txTail) blockHeader(rnd basics.Round) (bookkeeping.BlockHeader, bool) {
 	t.tailMu.RLock()
 	defer t.tailMu.RUnlock()
 	hdr, ok := t.blockHeaderData[rnd]
-	if !ok {
```

Comment:
We are removing this because now we have 2 types of callers to this function:
1. expect this function to succeed, and if not, they will handle (maybe log) the error. examples: `accountUpdatesLedgerEvaluator.BlockHdrCached`, 
2. expect the function to fail, in that case, they do a fallback without any error. example: `Ledger. BlockHdr`. In this case, logging is not needed, in fact it is wrong to log. Hence, the logging is removed.